plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkExtension.kt

```diff
@@ -9,7 +10,39 @@ open class JLinkExtension
 internal constructor(
         project: Project
 ) {
-    val configure: NamedDomainObjectCollection<JLinkOptions> = project.container(JLinkOptions::class.java) { name ->
+    val configure: NamedDomainObjectContainer<JLinkOptions> = project.container(JLinkOptions::class.java) { name ->
         JLinkOptions(name = name)
     }
-}
\ No newline at end of file
+
+    /**
+     * Registers a new jlink configuration.
+     *
+     * ```
+     * jlink {
+     *   "example" {
+     *     // options...
+     *   }
+     * }
+     * ```
+     *
+     * is equivalent to
+     * ```
+     * jlink {
+     *   register("example") {
+     *     // options...
+     *   }
+     * }
+     * ```
+     */
+    operator fun String.invoke(configurationAction: Action<in JLinkOptions>) = register(this, configurationAction)
+
+    /**
+     * Registers a new jlink configuration.
+     *
+     * @param name                the name of the configuration
+     * @param configurationAction the action to use to set up the configuration
+     */
+    fun register(name: String, configurationAction: Action<in JLinkOptions>)
+            = configure.register(name, configurationAction)
```

Comment:
Same question here.

Reply:
I don't see how this isn't usable from a Groovy DSL